TrainingApp/src/components/Input/Input.js

```diff
@@ -0,0 +1,83 @@
+import React, { useReducer, useEffect } from 'react';
+import { View, Text, TextInput } from 'react-native';
+import { INPUT_BLUR, INPUT_CHANGE } from '../../store/actions/actionTypes';
+import styles from './style'
+
+const inputReducer = (state, action) => {
+  switch (action.type) {
+    case INPUT_CHANGE:
+      return {
+        ...state,
+        value: action.value,
+        isValid: action.isValid
+      };
+    case INPUT_BLUR:
+      return {
+        ...state,
+        touched: true
+      };
+    default:
+      return state;
+  }
+};
+
+const Input = props => {
+  const [inputState, dispatch] = useReducer(inputReducer, {
+    value: props.initialValue ? props.initialValue : '',
+    isValid: props.initiallyValid,
```

Comment:
always de-structure here like
const { initialValue, initiallyValid }= props

Reply:
destructed all the props